File: Ex-21/ex_21_java/Departamento.java
public class Departamento {
    private String nome;
    // A referência para a universidade ainda existe
    private Universidade universidade;

    /**
     * Construtor para criar um Departamento de forma independente.
     * A universidade será associada depois.
     */
    public Departamento(String nome) {
        this.nome = nome;
        this.universidade = null; // Começa sem universidade afiliada
    }

    public String getNome() {
        return this.nome;
    }

    public Universidade getUniversidade() {
        return this.universidade;
    }
    
    /**
     * Método para definir/alterar a universidade a qual o departamento está filiado.
     * Este é o método chave para estabelecer a agregação.
     */
    public void setUniversidade(Universidade universidade) {
        this.universidade = universidade;
    }
}

File: Ex-21/c++/Departamento.cpp
#include "Departamento.hpp"

Departamento::Departamento() : nome(""),
                               universidade(nullptr){

                               };

Departamento::Departamento(const string &nome, Universidade *universidade) : nome(nome), universidade(universidade){

                                                                                         };
Departamento::~Departamento(){

};

void Departamento::setNome(const string &nome)
{
    this->nome = nome;
};

const string &Departamento::getNome() const
{
     return this->nome;
};

void Departamento::setUniversidade(Universidade *universidade)
{
     this->universidade = universidade;
};

const Universidade *Departamento::getUniversidade() const
{
    return this->universidade;
};